src/index.rs

```
@@ -1133,4 +1106,226 @@ mod tests {
+                file_name.push_str(&i.to_string());
+
+                let mut file_path = path.clone();
+                let mut new_file_path = path.clone();
+                file_path.push(file_name.clone());
+
+                let rand_new_filename: String = new_rand_filename();
+                new_file_path.push(rand_new_filename.clone());
+
+                let old_id: ResourceId;
+                match CanonicalPathBuf::canonicalize(&file_path) {
+                    Ok(canonicalized_path) => {
+                        old_id = initial_index.path2id
+                            [&canonicalized_path.clone()]
+                            .id;
+                    }
+                    Err(_) => {
+                        log::warn!(
+                            "File {} not found",
+                            file_path.to_str().unwrap_or("no_path_buf")
+                        );
+                        continue;
+                    }
+                }
+
+                let update_state = generate_random_update(
+                    path.clone(),
+                    Some(&file_name),
+                    Some(&rand_new_filename),
+                );
+
+                match update_state.try_into() {
+                    Ok(FileAction::CREATE) => {
+                        let _ = index1.track_addition(&new_file_path);
+                    }
+                    Ok(FileAction::UPDATE) => {
+                        let _ = index1.track_update(&file_path, old_id);
+                    }
+                    Ok(FileAction::DELETE) => {
+                        let _ = index1.track_deletion(old_id);
+                    }
+                    Ok(FileAction::MOVE) => {
+                        let _ = index1.track_deletion(old_id);
+                        let _ = index1.track_addition(&new_file_path);
+                    }
+                    Err(_) => println!("rnd_num error"),
+                }
```

Are we intentionally ignoring errors here for testing purposes?


If not, we should panic with errors

```suggestion
               match update_state.try_into() {
                    Ok(FileAction::CREATE) => {
                        index1
                            .track_addition(&new_file_path)
                            .expect("Should track addition correctly");
                    }
                    Ok(FileAction::UPDATE) => {
                        index1
                            .track_update(&file_path, old_id)
                            .expect("Should track update correctly");
                    }
                    Ok(FileAction::DELETE) => {
                        index1
                            .track_deletion(old_id)
                            .expect("Should track deletion correctly");
                    }
                    Ok(FileAction::MOVE) => {
                        index1
                            .track_deletion(old_id)
                            .expect("Should track deletion correctly");
                        index1
                            .track_addition(&new_file_path)
                            .expect("Should track addition correctly");
                    }
                    Err(_) => println!("rnd_num error"),
                }
```

---

@tareknaser yes, we must panic and not ignore the errors. But maybe simple unwrapping would be fine in test code?